airflow/models.py

```diff
@@ -3367,6 +3367,36 @@ def val(cls):
 
     @classmethod
     @provide_session
+    def setdefault(cls, key, default, deserialize_json=False, session=None):
+        """
+        Like a Python builtin dict object, setdefault returns the current value
+        for a key, and if it isn't there, stores the default value and returns it.
+
+        :param key: Dict key for this Variable
+        :type key: String
+        :param: default: Default value to set and return if the variable
+        isn't already in the DB
+        :type: default: Mixed
+        :param: deserialize_json: Store this as a JSON encoded value in the DB
+         and un-encode it when retrieving a value
+        :return: Mixed
+        """
+        obj = session.query(cls).filter(cls.key == key).first()
```

Comment:
I actually changed this again to completely use Variable.get and Variable.set.

Basically, I pass in an Object as the default with `deserialize_json=False`, and if I get the Object back, I KNOW it didn't find a value, in which case I set it and return it, otherwise, I decode the value, if the `deserialize_json=True`, and return it:

```
    @classmethod
    def setdefault(cls, key, default, deserialize_json=False):
        """
        Like a Python builtin dict object, setdefault returns the current value
        for a key, and if it isn't there, stores the default value and returns it.

        :param key: Dict key for this Variable
        :type key: String
        :param: default: Default value to set and return if the variable
        isn't already in the DB
        :type: default: Mixed
        :param: deserialize_json: Store this as a JSON encoded value in the DB
         and un-encode it when retrieving a value
        :return: Mixed
        """
        false_obj = {}
        obj = Variable.get(key, default_var=false_obj, deserialize_json=False)
        if obj == false_obj:
            if default is not None:
                Variable.set(key, default, serialize_json=deserialize_json)
                return default
            else:
                raise ValueError('Default Value must be set')
        else:
            if deserialize_json:
                return json.loads(obj.val)
            else:
                return obj.val

```